### PR TITLE
Set doctrine.orm.entity_manager.abstract as lazy

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Related to https://github.com/symfony/symfony/pull/19203

### CODE REVIEW

File: Resources/config/orm.xml [hunk 1]
```diff
@@ -95,7 +95,7 @@
 
         <service id="doctrine.orm.configuration" class="%doctrine.orm.configuration.class%" abstract="true" public="false" />
 
-        <service id="doctrine.orm.entity_manager.abstract" class="%doctrine.orm.entity_manager.class%" abstract="true" />
+        <service id="doctrine.orm.entity_manager.abstract" class="%doctrine.orm.entity_manager.class%" abstract="true" lazy="true" />
```
Comment (xabbuh) [inline Resources/config/orm.xml]: You may want to set the attribute dynamically in the extension class depending on whether or not it is available.

Comment (nicolas-grekas) [inline Resources/config/orm.xml]: Any idea how you'd do that? This is purely an implementation detail that doesn't leak through the interfaces. Given that next versions are going to need this "lazy", adding the dynamic check would only be a micro optim for soon to be legacy versions. Worth it?

Comment (Ocramius) [inline Resources/config/orm.xml]: > You may want to set the attribute dynamically in the extension class depending on whether or not it is available.

"lazy" is simply no-op if the required components aren't installed.

Comment (stof) [inline Resources/config/orm.xml]: I agree with @nicolas-grekas and @Ocramius. there is no need to make it conditional as it is already handled gracefully.

**Comment by stof** [comment]
you need to change the DI min dependency from 2.2 to 2.3 to fix the deps=low tests

File: Resources/config/orm.xml (hunk 1 above)
Comment (xabbuh) [inline Resources/config/orm.xml]: I am okay with that too if you then update the min version for the DependencyInjection component instead.

**Comment by nicolas-grekas** [comment]
@stof updated (line for DI added)

**Comment by kimhemsoe** [comment]
Thanks @nicolas-grekas

**Comment by antanas-arvasevicius** [comment]
Hello,
I've noticed that this change will trigger exception in our project:
  [Symfony\Component\DependencyInjection\Exception\ServiceCircularReferenceException]  
  Circular reference detected for service "doctrine.dbal.default_connection",          
   path: "doctrine.dbal.default_connection".  

We are using lazy:true in services.yml. Looks like your change is disabling other services which has lazy=true; Strange. Half of a day I'm searching for a problem, but can't find, maybe you could find why that happens. Could be bug in dependency container?
Thanks.

Test case:

services.yml:
  test:
    class: Test
    arguments: ['@doctrine.orm.default_entity_manager']
    lazy: true
    tags:
      - { name: doctrine.event_listener, event: prePersist }

test.php:
class Test
{
    public function __construct(EntityManager $em)
    {
    }
}

composer.json:
        "symfony/proxy-manager-bridge": "~2.3",
        "ocramius/proxy-manager": "~1.0",
        "symfony/symfony": "~2.8.9",

**Comment by kimhemsoe** [comment]
@antanas-arvasevicius Could you repost in #559 for now. It may/may not be the same error, but for now it is nice to all the information in the same issue until we know more.

**Comment by antanas-arvasevicius** [comment]
Hello, what did you mean by repost in #559 as this thread is #559?

By my example error is clearly known, it's due circular dependency of entity manager in listener service, and that's why I've had made that listener as a lazy service. 
But looks like this commit will break "lazy" support in the system. Or maybe DI won't allow lazy -> lazy -> lazy dependencies.

**Comment by kimhemsoe** [comment]
Sorry i meant #562

**Pull request merged**